bundles/net.tourbook/src/net/tourbook/ui/views/TourBlogView.java

```diff
@@ -344,37 +362,104 @@ private String buildDescription(String tourDescription) {
       final UrlDetector parser = new UrlDetector(tourDescription, UrlDetectorOptions.JAVASCRIPT);
       final List<Url> detectedUrls = parser.detect();
 
-      if (detectedUrls.isEmpty()) {
+      if (!detectedUrls.isEmpty()) {
+
+         // Because the tour description can contain similar URLs, this can create
+         // an issue when URLs can be replaced several times when replacing original
+         // text to formatted URLs.
+         // To avoid this, original URLs are first replaced by a unique random string
+         // and then this string is replaced by the formatted URL
+         final HashMap<String, Url> detectedUrlMap = new HashMap<>();
+         for (final Url detectedUrl : detectedUrls) {
+
+            final String randomString = RandomStringUtils.random(10, true, true);
+            detectedUrlMap.put(randomString, detectedUrl);
+
+            final String originalUrl = detectedUrl.getOriginalUrl();
+            tourDescription = StringUtils.replaceOnce(tourDescription, originalUrl, randomString);
+         }
+
+         for (final Map.Entry<String, Url> set : detectedUrlMap.entrySet()) {
+
+            final String fullUrl = set.getValue().getFullUrl();
+            tourDescription = tourDescription.replace(set.getKey(), "<a href=\"" + fullUrl + "\">" + fullUrl + "</a>"); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+
+         }
+      }
+
+      if (UI.IS_SCRAMBLE_DATA) {
+         tourDescription = UI.scrambleText(tourDescription);
+      }
+
+      return "<p class='description'>" + WEB.convertHTML_LineBreaks(tourDescription) + "</p>" + NL; //$NON-NLS-1$ //$NON-NLS-2$
+   }
+
+   private String buildTagsSection(final Set<TourTag> tourTags, final boolean addSpacer) {
+
+      final boolean showTourTags = Util.getStateBoolean(_state, TourBlogView.STATE_IS_SHOW_TOUR_TAGS, TourBlogView.STATE_IS_SHOW_TOUR_TAGS_DEFAULT);
+      if (!showTourTags) {
          return UI.EMPTY_STRING;
       }
 
-      // Because the tour description can contain similar URLs, this can create
-      // an issue when URLs can be replaced several times when replacing original
-      // text to formatted URLs.
-      // To avoid this, original URLs are first replaced by a unique random string
-      // and then this string is replaced by the formatted URL
-      final HashMap<String, Url> detectedUrlMap = new HashMap<>();
-      for (final Url detectedUrl : detectedUrls) {
+      final StringBuilder sb = new StringBuilder();
+
+      if (addSpacer) {
+         // write spacer
+         sb.append("<div>&nbsp;</div>");//$NON-NLS-1$
+      }
+
+      sb.append("<div class='title'>" + Messages.tour_editor_label_tour_tag + "</div>" + NL); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
When there are no tags, then this tag section should display nothing

Reply:
Fixed